flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/reader/KinesisStreamsSourceReader.java

```diff
@@ -64,6 +66,8 @@ public KinesisStreamsSourceReader(
     @Override
     protected void onSplitFinished(Map<String, KinesisShardSplitState> finishedSplitIds) {
         finishedSplitIds.keySet().forEach(this::unregisterShardMetricGroup);
+        context.sendSourceEventToCoordinator(
+                new SplitsFinishedEvent(new HashSet<>(finishedSplitIds.keySet())));
     }
```

Comment:
nit: can we swap the order? It would be better to send `SplitsFinishedEvent`, but miss unregister metric group, compared to the alternative.